### ISSUE TITLE
Feature: ability to open AVIF/HEIF files

### ISSUE THREAD

**Issue by GLLM** [hanatos/vkdt#60]
Hello !

This was to be requested sooner or later !
Can we please have AVIF/HEIF format compatibility in vkdt ?

More and more cameras support it, no later than today was Fuji X-T5 announced, which supports it !

Samples if need be (cant attach them). The author made them publicly available (see descriptions)

https://www.flickr.com/photos/michael_dittrich/50939561216/
https://www.flickr.com/photos/michael_dittrich/50709286547/

Thanks in advance
GLLM

**Comment by hanatos** [comment]
right. i don't have such a camera, and probably jxl would be an interesting format too. i should probably put some more documentation up on how to support file io.

# raw io modules

a good starting point is

https://github.com/hanatos/vkdt/tree/master/src/pipe/modules/i-pfm

for file input and

https://github.com/hanatos/vkdt/tree/master/src/pipe/modules/o-pfm

for file output (that's the modules to put in a graph).

# load image types in a folder by default

to support the file format with a default graph construction (such as the case for raw, mp4, jpg, ..) the extension will have to be registered and connected to a default input module like here:

https://github.com/hanatos/vkdt/blob/master/src/pipe/graph-defaults.h

which will then select default cfg from here (example for `i-jpg`, replace by appropriate module name):

https://github.com/hanatos/vkdt/blob/master/bin/default-darkroom.i-jpg
(default darkroom mode config file)

https://github.com/hanatos/vkdt/blob/master/bin/default.i-jpg
(default thumbnail processing config file)

# enable file format on export from gui

https://github.com/hanatos/vkdt/blob/master/src/gui/render_lighttable.cc#L277
https://github.com/hanatos/vkdt/blob/master/src/gui/render_lighttable.cc#L594

# enable file format from cli

`--format <o-modname>` might just work.
